test/unit/lib/plugins/aws/package/compile/functions.test.js

```diff
@@ -931,6 +931,40 @@ describe('AwsCompileFunctions', () => {
         ).to.deep.equal(compiledFunction);
       });
     });
+
+    it('should set function SnapStart ApplyOn to PublishedVersions when enabled', () => {
+      awsCompileFunctions.serverless.service.functions = {
+        func: {
+          handler: 'func.function.handler',
+          name: 'new-service-dev-func',
+          snapStart: true,
+        },
+      };
+
+      return expect(awsCompileFunctions.compileFunctions()).to.be.fulfilled.then(() => {
+        expect(
+          awsCompileFunctions.serverless.service.provider.compiledCloudFormationTemplate.Resources
+            .FuncLambdaFunction.Properties.SnapStart
+        ).to.deep.equal({ ApplyOn: 'PublishedVersions' });
+      });
+    });
+
+    it('should set function SnapStart ApplyOn to None when disabled', () => {
+      awsCompileFunctions.serverless.service.functions = {
+        func: {
+          handler: 'func.function.handler',
+          name: 'new-service-dev-func',
+          snapStart: false,
+        },
+      };
+
+      return expect(awsCompileFunctions.compileFunctions()).to.be.fulfilled.then(() => {
+        expect(
+          awsCompileFunctions.serverless.service.provider.compiledCloudFormationTemplate.Resources
+            .FuncLambdaFunction.Properties.SnapStart
+        ).to.deep.equal({ ApplyOn: 'None' });
+      });
+    });
```

Comment:
Let's write tests with `runServerless` util only (see second part of this test file, and https://github.com/serverless/serverless/tree/main/test#unit-tests